docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
+
+**Step3.** The parse method of AddProjectCommandParser is called with `commands` (the user input) as its parameter.
+
+**Step4.** AddProjectCommandParser will self call its `getProjectTitle` method. This will split the user's input and return the correct name for the project
+
+**Step5.** After returning this projectTitle to itself, `AddProjectCommandParser` will call the constructor for a new `AddProjectCommand` object
+
+**Step6.** This `AddProjectCommand` object will use the newly gathered project name to call the `addProjct` command of the `ProjectList` class.
+
+**Step7.** The status of this command completing is returned to `CommandHandler`, and the output is sent to the user.
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
+
+**Step1.** When `CommandHandler` receives a user input starting with string "listproject" or "listprojects", it will call `listProjects`. This method ensure that `projectList` has been initialized before the next step.
+
+**Step2.** Once the first step is done, `listProjects` will call the `printProject` method from `ProjectList` with no parameter.
+
+**Step3.** `printList` will first check if the size of the private ArrayList named `projectList` is 0, which means that the arraylist is empty. If yes, it will print "You have not added any projects yet!" and return.
+
+**Step4.** It will then iterate over `projectList`, which type is `Project`. It will call `getTitle` and `getDeadline` from each array element in`the `Project` class.
+
+**Step5.** the general information of a project, i.e., its title and deadline, will be displayed.
+
+#### Todo feature
+![image info](./UmlDiagrams/Todo.png)  
+
+**Step1.** When `CommandHandler` receives a user input starting with string “todo”, it will call `toDo()` function to parse the user input into the index of the target `Project` object and one description string.  
+
+**Step2.** If the number of provided arguments is less than the required number, the `toDo()` function will throw an `IllegalCommand` exception and return.  
+
+**Step3.** If no exception is thrown, two strings will be generated by parsing the user input and the `toDo()` function will pass the two arguments to `projectList` for creating and adding a `Todo` object to a `Project` object.  
+
+**Step4.** When `projectList` receives the call, it will get the target `Project` object and pass the description string to it.  
+
+**Step5.** The `Project` object will create a `Todo` object and add it to the `ArrayList<Todo>` in `Project` class.  
+
+#### View a Project
```

Comment:
Would be good if you all can decide whether to put images before steps or steps before images